doc/rtd/development/datasource_creation.rst

```diff
@@ -170,6 +170,8 @@ Datasources included in upstream cloud-init benefit from ongoing maintenance,
 compatibility with the rest of the codebase, and security fixes by the upstream
 development team.
 
+If this is not possible, one can add
+:ref:`custom out-of-tree datasources to cloud-init<custom_datasource>`.
```

Comment:
nit: pull cloud-init out of this link text (aligns w/ how we setup the link for custom modules in the other file)